USER_GUIDE.md

```diff
@@ -0,0 +1,123 @@
+- [Setup](#setup)
+- [Sample code](#sample-code)
+- [AWS Managed Opensearch with Sigv4](#aws-managed-opensearch-with-sigv4)
+
+## Setup
+
+To add the client to your project, install it using [RubyGem](https://rubygems.org/)
+
+`gem install opensearch-ruby`
+
+or, install it from a source code checkout:
+```bash
+git clone https://github.com/opensearch-project/opensearch-ruby.git
+cd opensearch-ruby/opensearch
+gem build opensearch.gemspec
+gem install opensearch-ruby-<version>.gem
+```
```

Comment:
99% of people in the Ruby world use bundler, so I think instead of all this we should just say "add `gem opensearch-ruby`" to your Gemfile and run `bundle install`.